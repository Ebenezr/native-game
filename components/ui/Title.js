import { StyleSheet, Text } from 'react-native';
import Colors from '../../constants/colors';

const Title = ({ title }) => {
  return <Text style={styles.title}>{title}</Text>;
};

const styles = StyleSheet.create({
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: Colors.white,
    textAlign: 'center',
    borderWidth: 2,
    borderColor: Colors.white,
    padding: 12,
  },
});

export default Title;
