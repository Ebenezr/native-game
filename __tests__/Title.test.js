import renderer from 'react-test-renderer';
import Title from '../components/ui/Title';

describe('Title', () => {
  it('renders Title correctly', () => {
    const tree = renderer
      .create(<Title title='Test Title'>Test Children</Title>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
