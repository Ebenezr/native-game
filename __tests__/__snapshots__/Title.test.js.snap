// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders Card correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#72063c",
      "borderRadius": 10,
      "elevation": 5,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginTop": 40,
      "padding": 20,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 6,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Test Title
  </Text>
  Test Children
</View>
`;

exports[`Title renders Title correctly 1`] = `
<Text
  style={
    {
      "borderColor": "#fff",
      "borderWidth": 2,
      "color": "#fff",
      "fontFamily": "open-sans-bold",
      "fontSize": 24,
      "padding": 12,
      "textAlign": "center",
    }
  }
>
  Test Title
</Text>
`;
